src/components/SendTokens/ETHTokens.js

```diff
@@ -0,0 +1,495 @@
+import { fontStyles, spacing, UIColors } from 'utils/variables';
+import { getBalance, getRate, calculateMaxAmount, checkIfEnoughForFee } from 'utils/assets';
+import { makeAmountForm, getAmountFormFields } from 'utils/formHelpers';
+import { checkIfSmartWalletAccount } from 'utils/accounts';
+import { calculateGasEstimate } from 'services/assets';
+import smartWalletService from 'services/smartWallet';
+
+// types
+import type { NavigationScreenProp } from 'react-navigation';
+import type { GasInfo } from 'models/GasInfo';
+import type { TokenTransactionPayload } from 'models/Transaction';
+import type { Balances, Rates } from 'models/Asset';
+import type { RootReducerState } from 'reducers/rootReducer';
+
+// constants
+import { SEND_TOKEN_CONFIRM } from 'constants/navigationConstants';
+import { ETH, defaultFiatCurrency, SPEED_TYPES } from 'constants/assetsConstants';
+
+// actions
+import { fetchGasInfoAction } from 'actions/historyActions';
+import { updateAppSettingsAction } from 'actions/appSettingsActions';
+
+// selectors
+import type { Account } from 'models/Account';
+import Spinner from 'components/Spinner';
+
+const ActionsWrapper = styled.View`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+`;
+
+const SendTokenDetailsValue = styled(BaseText)`
+  ${fontStyles.medium};
+`;
+
+const ButtonWrapper = styled.View`
+  margin-top: ${spacing.rhythm / 2}px;
+  margin-bottom: ${spacing.rhythm + 10}px;
+`;
+
+const Btn = styled(Button)`
+  margin-top: 14px;
+  display: flex;
+  justify-content: space-between;
+`;
+
+const FooterInner = styled.View`
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: flex-end;
+  width: 100%;
+  padding: ${spacing.large}px;
+  background-color: ${UIColors.defaultBackgroundColor};
+`;
+
+const BackgroundWrapper = styled.View`
+  background-color: ${UIColors.defaultBackgroundColor};
+  flexGrow: 1;
+`;
+
+type Props = {
+  assetData: Object,
+  receiver: string,
+  source: string,
+  token: string;
+  address: string,
+  totalBalance: number,
+  contractAddress: string,
+  navigation: NavigationScreenProp<*>,
+  isVisible: boolean,
+  formValues?: Object,
+  balances: Balances,
+  session: Object,
+  fetchGasInfo: Function,
+  gasInfo: GasInfo,
+  rates: Rates,
+  baseFiatCurrency: ?string,
+  transactionSpeed: ?string,
+  updateAppSettings: Function,
+  activeAccountAddress: string,
+  activeAccount: ?Account,
+};
+
+type State = {
+  value: ?{
+    amount: ?string,
+  },
+  showModal: boolean,
+  gasLimit: number,
+  gettingFee: boolean,
+  calculatingMaxValue: boolean,
+  inputHasError: boolean,
+  txFeeInWei: BigNumber,
+  submitPressed: boolean,
+};
+
+const { Form } = t.form;
+const MIN_TX_AMOUNT = 0.000000000000000001;
+
+const SPEED_TYPE_LABELS = {
+  [SPEED_TYPES.SLOW]: 'Slow',
+  [SPEED_TYPES.NORMAL]: 'Normal',
+  [SPEED_TYPES.FAST]: 'Fast',
+};
+
+class SendTokenAmount extends React.Component<Props, State> {
```

Comment:
I would change the class name as well. Could be `SendETHTokens` as in your reference, or `ETHTokens` as in file name, or something similar

Reply:
Fixed.